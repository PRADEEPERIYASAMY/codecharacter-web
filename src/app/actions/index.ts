export * from 'app/actions/dashboard';
