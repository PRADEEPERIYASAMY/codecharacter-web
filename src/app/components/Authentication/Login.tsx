import { faEnvelope, faLock } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { AuthType } from 'app/components/Authentication/';
import * as styles from 'app/styles/Authentication.module.css';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Row } from 'react-bootstrap';

export class Login extends React.Component<Login.Props, Login.State> {
  private loginRef = React.createRef<HTMLFormElement>();

  constructor(props: Login.Props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  public render() {
    return (
      <div>
        <Row>
          <div className={classnames('col-sm-10 offset-sm-1', styles.form)}>
            <form
              className={classnames('loginForm')}
              noValidate
              ref={this.loginRef}
              onSubmit={(e) => {
                const form = this.loginRef.current;
                if (form) {
                  if (!form.checkValidity()) {
                    e.preventDefault();
                  }
                  form.classList.add('was-validated');
                }
              }}
            >
              <div className="form-row">
                <div className="col mb-4">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faEnvelope} />
                      </span>
                    </div>
                    <input
                      type="email"
                      className="form-control"
                      id="validationEmail"
                      placeholder="Your email"
                      aria-describedby="inputGroupPrepend"
                      required
                      value={this.state.email}
                      onChange={(e) =>
                        this.setState({
                          email: e.target.value,
                        })
                      }
                    />
                    <div className="invalid-feedback">Please enter a valid Email ID.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-4">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faLock} />
                      </span>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="validationPassword"
                      placeholder="Your password"
                      aria-describedby="inputGroupPrepend"
                      minLength={5}
                      required
                    />
                    <div className="invalid-feedback">Please enter the correct password.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col text-center">
                  <button
                    className="btn btn-info"
                    type="submit"
                    style={{
                      width: '100%',
                    }}
                  >
                    Login
                  </button>
                </div>
              </div>
            </form>
          </div>
        </Row>
        <Row
          style={{
            borderTop: '2px solid #999',
          }}
        >
          <Col className="text-center my-3 ml-auto mr-auto">
            <div className="text-dark">
              Don't have an account?{' '}
              <a
                className="text-primary"
                style={{
                  cursor: 'pointer',
                }}
                onClick={() => this.props.handleSelectPanel(AuthType.REGISTER)}
              >
                Create one
              </a>
            </div>
          </Col>
        </Row>
      </div>
    );
  }
}

export namespace Login {
  export interface OwnProps {
    handleSelectPanel: (authType: AuthType) => void;
  }
  export interface State {
    email: string;
    password: string;
  }
  export type Props = OwnProps;
}
