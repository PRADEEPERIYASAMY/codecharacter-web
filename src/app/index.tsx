import { Dashboard } from 'app/components/Dashboard';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
