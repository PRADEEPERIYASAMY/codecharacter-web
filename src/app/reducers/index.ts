import { codeReducer } from 'app/reducers/code/Code';
import { editorReducer } from 'app/reducers/code/Editor';
import { dashboardReducer } from 'app/reducers/Dashboard';
import { gameLogReducer } from 'app/reducers/GameLog';
import { leaderboardReducer } from 'app/reducers/Leaderboard';
import { notificationReducer } from 'app/reducers/Notification';
import { userReducer } from 'app/reducers/User';
import * as CodeInterfaces from 'app/types/code/Code';
import * as EditorInterfaces from 'app/types/code/Editor';
import * as DashboardInterfaces from 'app/types/Dashboard';
import * as GameLogInterfaces from 'app/types/GameLog';
import * as LeaderboardInterfaces from 'app/types/Leaderboard';
import * as UserInterfaces from 'app/types/User';
import { routerReducer, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';

export const rootReducer = combineReducers({
  code: codeReducer,
  dashboard: dashboardReducer,
  editor: editorReducer,
  gameLog: gameLogReducer,
  leaderboard: leaderboardReducer,
  notification: notificationReducer,
  router: routerReducer,
  user: userReducer,
});

export interface RootState {
  code: CodeInterfaces.CodeStoreState;
  dashboard: DashboardInterfaces.DashboardStoreState;
  editor: EditorInterfaces.EditorStoreState;
  leaderboard: LeaderboardInterfaces.LeaderboardStoreState;
  router: RouterState;
  gameLog: GameLogInterfaces.GameLogStoreState;
  user: UserInterfaces.UserStoreState;
}
