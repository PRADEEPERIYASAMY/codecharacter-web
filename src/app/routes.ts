export enum Routes {
  ROOT = '/',
  LOGIN = '/login',
  REGISTER = '/register',
  LEADERBOARD = '/leaderboard',
  USER_PROFILE_MODEL = '/profile',
<<<<<<< HEAD
  GITHUB_OAUTH = '/login/github',
  GOOGLE_OAUTH = '/login/google',
  USER_ACTIVATION = '/user-activate',
=======
  PROFILE_USER_STATS = '/profile/:username',
>>>>>>> Add route for /profile/:user and Add the required container
}
