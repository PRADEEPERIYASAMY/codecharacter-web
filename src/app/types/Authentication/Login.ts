import { AuthType } from 'app/types/Authentication';
export { Login as OwnState } from 'app/types/User';

export interface OwnProps {
  handleSelectPanel: (authType: AuthType) => void;
}

export interface DispatchProps {
  login: (username: string, password: string) => void;
}

export type Props = OwnProps & DispatchProps;
