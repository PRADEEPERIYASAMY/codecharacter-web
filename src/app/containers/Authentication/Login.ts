import { UserActions } from 'app/actions';
import { Login } from 'app/components/Authentication/Login';
import * as LoginInterfaces from 'app/types/Authentication/Login';
import { connect } from 'react-redux';

const loginContainer = connect<LoginInterfaces.Props, LoginInterfaces.DispatchProps, {}>(
  null,
  {
    login: UserActions.login,
  },
)(Login);

export default loginContainer;
