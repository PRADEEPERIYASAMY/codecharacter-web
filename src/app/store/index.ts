import { logger } from 'app/middleware';
import { rootReducer } from 'app/reducers';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

export function configureStore(initialState?: object) {
  let middleware = applyMiddleware(logger);

  if (process.env.NODE_ENV !== 'production') {
    middleware = composeWithDevTools(middleware);
  }

  const store = createStore(rootReducer, initialState, middleware);

  if (module.hot) {
    module.hot.accept('app/reducers', () => {
      const nextReducer = require('app/reducers');
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
