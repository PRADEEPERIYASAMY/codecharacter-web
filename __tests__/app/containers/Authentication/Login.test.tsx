import Login from 'app/containers/Authentication/Login';
import { configureStore } from 'app/store';
import { mount } from 'enzyme';
import * as React from 'react';

describe('Login Container', () => {
  const store = configureStore();
  // tslint:disable-next-line:no-console
  console.log(<Login handleSelectPanel={Function} />);
  const wrapper = mount(<Login handleSelectPanel={Function} />, {
    context: {
      store,
    },
  });

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
