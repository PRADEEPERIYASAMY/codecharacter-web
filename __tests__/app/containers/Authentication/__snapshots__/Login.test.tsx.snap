// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container Should render 1`] = `
<Connect(Login)
  handleSelectPanel={[Function]}
>
  <Login
    handleSelectPanel={[Function]}
    login={[Function]}
  >
    <div>
      <Row
        bsClass="row"
        componentClass="div"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-10 offset-sm-1"
          >
            <form
              className="loginForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-row"
              >
                <div
                  className="col mb-4"
                >
                  <div
                    className="input-group"
                  >
                    <div
                      className="input-group-prepend"
                    >
                      <span
                        className="input-group-text"
                        id="inputGroupPrepend"
                      >
                        <FontAwesomeIcon
                          border={false}
                          className=""
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                448,
                                512,
                                Array [],
                                "f007",
                                "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                              ],
                              "iconName": "user",
                              "prefix": "fas",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-user fa-w-14 "
                            data-icon="user"
                            data-prefix="fas"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 448 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </span>
                    </div>
                    <input
                      aria-describedby="inputGroupPrepend"
                      className="form-control"
                      id="validationUsername"
                      onChange={[Function]}
                      placeholder="Username"
                      required={true}
                      type="text"
                      value=""
                    />
                    <div
                      className="invalid-feedback"
                    >
                      Please enter a valid Username.
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="form-row"
              >
                <div
                  className="col mb-4"
                >
                  <div
                    className="input-group"
                  >
                    <div
                      className="input-group-prepend"
                    >
                      <span
                        className="input-group-text"
                        id="inputGroupPrepend"
                      >
                        <FontAwesomeIcon
                          border={false}
                          className=""
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                448,
                                512,
                                Array [],
                                "f023",
                                "M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z",
                              ],
                              "iconName": "lock",
                              "prefix": "fas",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-lock fa-w-14 "
                            data-icon="lock"
                            data-prefix="fas"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 448 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </span>
                    </div>
                    <input
                      aria-describedby="inputGroupPrepend"
                      className="form-control"
                      id="validationPassword"
                      minLength={5}
                      placeholder="Password"
                      required={true}
                      type="password"
                    />
                    <div
                      className="invalid-feedback"
                    >
                      Please enter the correct password.
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="form-row"
              >
                <div
                  className="col text-center"
                >
                  <button
                    className="btn btn-info"
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                    type="submit"
                  >
                    Login
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </Row>
      <Row
        bsClass="row"
        componentClass="div"
        style={
          Object {
            "borderTop": "2px solid #999",
          }
        }
      >
        <div
          className="row"
          style={
            Object {
              "borderTop": "2px solid #999",
            }
          }
        >
          <Col
            bsClass="col"
            className="text-center my-3 ml-auto mr-auto"
            componentClass="div"
          >
            <div
              className="text-center my-3 ml-auto mr-auto"
            >
              <div
                className="text-dark"
              >
                Don't have an account?
                 
                <a
                  className="text-primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  Create one
                </a>
              </div>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Login>
</Connect(Login)>
`;
